python/TestHarness/schedulers/QueueManager.py

```diff
@@ -70,70 +77,57 @@ def __createSessionFile(self):
             session_file = open(self.options.session_file, 'w')
 
         except IOError:
-            raise QueueManagerError('Can not open %s for writing!' % (self.options.session_file))
+            print('Can not open %s for writing!' % (self.options.session_file))
+            sys.exit(1)
 
         return session_file
 
-    def __copyFiles(self, job_container, template):
+    def __copyFiles(self, job, template):
         """ Copy the necessary test files the job will need in order to execute """
-        tester = job_container.getTester()
-
-        # Create regexp object of no_copy_pattern
-        if 'no_copy_pattern' in self.params.keys():
-            # Match no_copy_pattern value
-            pattern = re.compile(self.params['no_copy_pattern'])
-        else:
-            # Match nothing if not set. Better way?
-            pattern = re.compile(r'')
+        tester = job.getTester()
 
         # Create the job directory
-        if not os.path.exists(self.getWorkingDir(job_container)):
+        if not os.path.exists(self.getWorkingDir(job)):
             try:
-                os.makedirs(self.getWorkingDir(job_container))
+                os.makedirs(self.getWorkingDir(job))
             except OSError, ex:
                 if ex.errno == errno.EEXIST: pass
                 else: raise
 
-        # Move into the newly created test directory
-        current_pwd = os.getcwd()
-        os.chdir(self.getWorkingDir(job_container))
+        source_dir = os.path.dirname(self.getWorkingDir(job))
+        target_dir = self.getWorkingDir(job)
 
-        # Copy files (unless they are listed in "no_copy")
-        for file in os.listdir('../'):
-            if os.path.isfile('../' + file) and file != self.options.input_file_name and \
-               (not tester.specs.isValid('no_copy') or file not in tester.specs['no_copy']) and \
-               (not 'no_copy_pattern' in self.params.keys() or pattern.match(file) is None) and \
-               not os.path.exists(file) and \
+        # Copy files
+        for file in os.listdir(source_dir):
+            if os.path.isfile(os.path.join(source_dir, file)) and file != self.options.input_file_name and \
+               not os.path.exists(os.path.join(target_dir, file)) and \
                os.path.splitext(file)[1] != '':
-                shutil.copy('../' + file, '.')
+                shutil.copy(os.path.join(source_dir, file), os.path.join(target_dir, file))
 
         # Copy directories
         if 'copy_files' in template.keys():
             for file in template['copy_files']:
-                if os.path.isfile('../' + file):
-                    if not os.path.exists(file):
-                        shutil.copy('../' + file, '.')
-                elif os.path.isdir('../' + file):
+                if os.path.isfile(os.path.join(source_dir, file)):
```

Comment:
So many joins... it might help readability if you just have a `source_file` and `target_file` where you do the `os.path.join` once.

Reply:
👍